generators/typescript/asIs/tests/mock-server/withHeaders.ts

```diff
@@ -0,0 +1,80 @@
+import { HttpResponseResolver, passthrough } from "msw";
+
+/**
+ * Creates a request matcher that validates if request headers match specified criteria
+ * @param expectedHeaders - Headers to match against
+ * @param resolver - Response resolver to execute if headers match
+ */
+export function withHeaders(
+    expectedHeaders: Record<string, string | RegExp | ((value: string) => boolean)>,
+    resolver: HttpResponseResolver
+): HttpResponseResolver {
+    return (args) => {
+        const { request } = args;
+        const { headers } = request;
+
+        // Track mismatched headers
+        const mismatches: Record<
+            string,
+            { actual: string | null; expected: string | RegExp | ((value: string) => boolean) }
+        > = {};
+
+        // Check each expected header
+        for (const [key, expectedValue] of Object.entries(expectedHeaders)) {
+            // Get the header value (case-sensitive)
```

Comment:
I think we should remove a lot of these comments since they more-or-less explain the obvious. Given that this is checked into each SDK, I'd rather only include comments where totally necessary (like the function signature comment you included at the top).